src/fa/game_session.py

```diff
@@ -94,6 +94,16 @@ def listen(self):
         else:
             self.connectivity.prepare()
 
+    def _needs_game_connection(fn):
+        def wrap(self, *args, **kwargs):
+            if self._game_connection is None:
+                self._logger.warn("{}.{}: tried to run without a game connection".format(
+                    self.__class__.__name__, fn.__name__))
+            else:
+                return fn(self, *args, **kwargs)
```

Comment:
wrap is a wrapper for fn which either does nothing or executes fn, so there is no sense for it to return fn (rather than execute it). The decorator takes a function (fn) and returns a wrapper, so it's fine as well.